python/CMakeLists.txt

```
@@ -23,12 +23,18 @@ else()
     )
 endif()
 
+install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/podio_gen
+  DESTINATION ${podio_PYTHON_INSTALLDIR}
+  REGEX test_.*\\.py$ EXCLUDE  # Do not install test files
+  PATTERN __pycache__ EXCLUDE  # Or pythons caches
+  )
+
 #--- install templates ---------------------------------------------------------
 install(DIRECTORY ${CMAKE_CURRENT_SOURCE_DIR}/templates
   DESTINATION ${podio_PYTHON_INSTALLDIR})
 
 IF (BUILD_TESTING)
-  add_test( NAME pyunittest COMMAND python3 -m unittest discover -s ${PROJECT_SOURCE_DIR}/python/podio)
+  add_test( NAME pyunittest COMMAND python3 -m unittest discover -s ${PROJECT_SOURCE_DIR}/python/podio -p "test_*.py")
```

```suggestion
  add_test( NAME pyunittest COMMAND python3 -m unittest discover -s ${PROJECT_SOURCE_DIR}/python -p "test_*.py")
```
I think otherwise we are missing the ones that are now in `podio_gen`, right?

---

Ah yes, of course, that is what I tested before and it doesn't work. I'm checking...

---

So I have no idea how to make it work with a single test. First, when trying to discover from the top directory python will call `dir(podio)` and will fail like in https://github.com/AIDASoft/podio/issues/512. Even finding a workaround some tests fail with errors related to some errors when importing like this
``` python
ImportError: Failed to import test module: podio.test_EventStore
Traceback (most recent call last):
  File "/usr/lib/python3.11/unittest/loader.py", line 407, in _find_test_path
    module = self._get_module_from_name(name)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/usr/lib/python3.11/unittest/loader.py", line 350, in _get_module_from_name
    __import__(name)
  File "/usr/lib/python3.11/site-packages/ROOT/_facade.py", line 154, in _importhook
    return _orig_ihook(name, *args, **kwds)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
ModuleNotFoundError: No module named 'podio.test_EventStore'; 'podio' is not a package
```
so I reverted back to the version that works except pre-commit...